design/Proposal For View Definition.md

```diff
@@ -0,0 +1,84 @@
+﻿
+
+
+
+## Proposal to extend the View Definition
+**Background**
+The current  interface for a SqlView  which is a first class object in Nessie specifies a Sql text and a Dialect  and schema :
+
+    public interface SqlView extends Contents {  
+      
+      public static enum Dialect {  
+        HIVE,  
+        SPARK,  
+        DREMIO,  
+        PRESTO  
+      }  
+      
+      String getSqlText();  
+      
+      Dialect getDialect();  
+      
+      // Schema getSchema();  
+    }
+
+While thinking of integrating Dremio with Nessie, we found we needed several more fields to fully define a view along with  a schema  (written up in a separate proposal) 
+
+**Proposal for additional fields**
+While a View is ultimately describe by just the SQL query , these  additional fields needed to describe a view (or what we refer to as a Virtual Dataset in the context of Dremio) for the DREMIO dialect . 
+
+   
+**Generic Vs Implementation specific View Definition**
+Some fields are  generic enough that  we could create a  definition common to view definitions of any dialect. Each "Dialect" can  then define its own extended fields needed for maintaining metadata for its internal implementation of a view (shown below for Dremio) .
+
+**Generic fields**
+
+    message PSqlView {
+    
+    required string sqlText = 1 ; // SQL string  that defines the view
+    
+    string dialect = 2 ; // Describes the type of system this view is defined on - eg hive, presto, dremio etc. 
+    
+    optional ViewSchema schema = 3; // Details on the fields/column of the view, the datatypes, precision and scale. (See definition of ViewSchema)
+    
+    repeated string context = 4; // Defines the folder, space the view is defined on. Represented as list of strings which are keys  to container objects in Nessie.Eg ["@myspace","myfolder","mysubfolder"] or ["mycatalog", "myschema"]
+       
+    }
+    
+
+**Dremio specific  fields** 
```

Comment:
Nessie trying to be an interoperable system, it doesn't seem like a good idea to have fields designed for a specific system in mind while other systems might know how to interpret them, or how to deal with them for certain operations. For example, let's say you have some tool allowing to copy an object under a different name, I guess reusing the same datasetId might become an issue?